oauth2_http/java/com/google/auth/oauth2/SecureSessionAgent.java

```diff
@@ -58,7 +58,7 @@
  * <p>This is an experimental utility.
  */
 @ThreadSafe
-public final class S2A {
+public class SecureSessionAgent {
```

Comment:
Can we keep final or does that break something?

Reply:
Can you point me to where we need to mock it? I think we should be able to find other ways around, but just want to check.

Reply:
In https://github.com/googleapis/sdk-platform-java/pull/3326 if we want to test: `createS2ASecuredChannelCredentials` function added. I was thinking Mockito would be the simplest way to mock this lib, in which case I believe the class would have to be non-final

Reply:
Do you have an example of how we're testing in the PR right now?

One option that comes to mind for testing this without mocks:
Update the method to: `createS2ASecuredChannelCredentials(SecureSessionAgent s2a)`

The tests for createS2ASecuredChannelCredentials would pass version of SecureSessionAgent (i.e. SecureSessionAgentTest which overrides the `getConfig()` and returns SecureSessionAgentConfig with the needed values). 

I *think* it's possible to test it without having to mock it.

Reply:
Ah ignore the above, I was mistaken. I am fine with removing the final keyword.